src/api/hooks/useGetANS.ts

```diff
@@ -43,23 +96,30 @@ export function useGetNameFromAddress(
     queryKey: ["ANSName", address, shouldCache, state.network_name],
     queryFn: () => {
       const standardizedAddress = standardizeAddress(address);
-      const knownName = knownAddresses[standardizedAddress.toLowerCase()];
-      if (knownName) {
-        return knownName;
-      }
 
       // Change cache key specifically to invalidate all previous cached keys
       const cachedName = getLocalStorageWithExpiry(`${address}:name`);
       if (cachedName) {
         return cachedName;
       }
       // Ensure there's always .apt at the end
-      return genANSName(
+      const ansName = genANSName(
         address,
         shouldCache,
         state.network_name,
         isValidator,
       ).then((name) => (name ? `${name}.apt` : null));
+
+      // use ANS name if available, otherwise use knownName
+      // ideally on account page, we show both
+      if (ansName == null) {
+        const knownName = knownAddresses[standardizedAddress.toLowerCase()];
+        if (knownName) {
+          return knownName;
+        } else {
+          return null;
+        }
+      }
```

Comment:
I'm up in the air here, I think known names should take precedence, because they can't be added from other users, where the ANS names can be added onto accounts.  (unless we're getting primary names)

Reply:
I was thinking more of the 'right' thing for ecosystem project is to buy an ANS name for themself but they'd be disincentivized if we don't even show it?

Reply:
The right thing is always to use the ANS name, it's easily used across the ecosystem.  However, some things may never get them (like exchange addresses).

Gotta check if you can override someone else's address (which was a thing at one point), in that case, we'd want to stick with the label

Reply:
I lean towards @ying-w's pov. @gregnazario We are also fetching primary names in this particular instance.

Reply:
Ah, if we are fetching primary names, we can roll with it.

I'll come up with a good way to display both anyways on the account page, I asked ying to revert for now, and I'll apply it

Reply:
I opened an issue for showing both https://github.com/aptos-labs/explorer/issues/849